features/all-positions/LiquidationActionDialog.tsx

```diff
@@ -0,0 +1,488 @@
+        };
+      } else if (liquidationStatus === "DisputeSucceeded") {
+        return {
+          status: "Dispute Succeeded",
+          description:
+            "The liquidation was disputed and the DVM deemed the liquidation invalid (valid dispute).",
+        };
+      } else if (liquidationStatus === "DisputeFailed") {
+        return {
+          status: "Dispute Failed",
+          description:
+            "The liquidation was disputed and the DVM deemed the liquidation valid (invalid dispute).",
+        };
+      } else {
+        return { status: "Unknown", description: "" };
+      }
+    };
+
+    const prettyDate = (x_secs: number) => {
+      return new Date(x_secs * 1000).toLocaleString("en-GB", {
+        timeZone: "UTC",
+      });
+    };
+
+    const getDisputablePrice = (x: string) => {
+      return invertDisputablePrice && parseFloat(x) !== 0
+        ? 1 / Number(x)
+        : Number(x);
+    };
+
+    const executeDispute = async () => {
+      if (!collBalanceTooLow) {
+        setHash(null);
+        setSuccess(null);
+        setError(null);
+
+        try {
+          if (needCollateralAllowance()) await setMaxCollateralAllowance();
+          const tx = await emp.dispute(
+            liquidatedPosition.liquidationId,
+            liquidatedPosition.sponsor
+          );
+          setHash(tx.hash as string);
+          await tx.wait();
+
+          setSuccess(true);
+        } catch (error) {
+          console.error(error);
+          setError(error);
+        }
+      } else {
+        setError(new Error("Please check that you are connected."));
+      }
+    };
+
+    return (
+      <Dialog open={props.isDialogShowing} onClose={props.handleClose}>
+        <PositionDialog>
+          <Container>
+            <Box>
+              <Box>
+                <h1>Liquidation Info & Actions</h1>
+                <Status>
+                  <Label>Liquidation status: </Label>
+                  <Tooltip
+                    title={
+                      translateLiquidationStatus(
+                        Number(liquidatedPosition.liquidationTimestamp),
+                        liquidatedPosition.status
+                      ).description
+                    }
+                    placement="top"
+                  >
+                    <span>
+                      {" "}
+                      {
+                        translateLiquidationStatus(
+                          Number(liquidatedPosition.liquidationTimestamp),
+                          liquidatedPosition.status
+                        ).status
+                      }
+                    </span>
+                  </Tooltip>
+                </Status>
+                <Status>
+                  <Label>Liquidation Timestamp (UTC): </Label>
+                  <Tooltip
+                    title={`Liquidation unixTimestamp: ${liquidatedPosition.liquidationTimestamp}`}
```

Comment:
Maybe should be more clear/less technical and say "Liquidated @" and somhow indicate that this is the timestamp whose price you will be disputing

Reply:
ye, can make it clearer. Take a look.